src/__tests__/index.test.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
```

Comment:
would u be willing to refactor this to use `react-testing-library` instead of enzyme?

Reply:
I have not used that one before, I looked over the documentation and it appears to be missing features such as prop/state modifying. Since this component relies on state modifications to run, I recommend a library like enzyme is used for testing it.

Reply:
that's kinda the point of using this library instead of enzyme, consumers of this component don't modify state directly, so I think the tests shouldnt either - every state change has to be triggered by some event or rerendering (due to props changing), such tests would give us more confidence than testing internal implementation

Reply:
Testing the internal implementation is going to give you much more confidence in making changes in the future, I've found that only testing user-facing functionality (and snapshots) makes it very easy to miss bugs. I can add tests to this that mount the component as well, to have a combination of unit and functional tests.

Reply:
I don't quite agree with you. It's a common opinion (with which I agree), that integration tests can give you way more confidence than unit tests (example read https://blog.kentcdodds.com/write-tests-not-too-many-mostly-integration-5e8c7fff591c).

It's ofc not always easy to write integration tests and I don't want you to write any here, but I feel like moving higher this spectrum ladder is better. And that by imitating the user and their interactions with the component would give us more confidence. Because of the nature of this component (high coupling to the DOM as we have to measure the width of the hidden textarea) I'm not entirely sure if this is going to be easy using jest+jsdom, but might be worth exploring.

This would still be in my eyes a unit test (unit being a component), I'd rather like to test its interfaces and outputs rather than internal implementation.

That's ofc my opinion and I respect yours. I'm also really thankful for this contribution ❤️ any tests are better than none. If you don't feel like working towards what I'm proposing then its OK, I'll be happy to merge the PR as is and creating a separate ticket about trying out migration to react-testing-library.